File: Level-1/40.java
class Solution {
    public int[] solution(int N, int[] stages) {
        int[] answer = new int[N];
        double[] result = new double[N];
        for(int i = 1; i <= N; i++){
            int n = 0;
            int m = 0;
            for(int j = 0; j < stages.length; j++){
                if(i == stages[j]) n += 1;
                if(i <= stages[j]) m += 1;
            }
            result[i - 1] = (double)n / m;
            answer[i - 1] = i;
        }
        
        for(int i = 0; i < N - 1; i++){
            for(int j = 0; j < N - 1 - i; j++){
                if(result[j] < result[j + 1]){
                    double temp = result[j + 1];
                    result[j + 1] = result[j];
                    result[j] = temp;
                    int temp2 = answer[j + 1];
                    answer[j + 1] = answer[j];
                    answer[j] = temp2;
                }
            }
        }
        
        return answer;
    }
}

File: Level-1/40.js
function solution(N, stages) {
    let fail = [];
    
    for(let i = 1; i <= N; i++) {
        let num = stages.filter(v => v === i).length;
        let num2 = stages.filter(v => v >= i).length;
        fail.push([i, num / num2]);
    }
    
    fail.sort((a, b) => b[1] - a[1]);
    
    return fail.map(v => v[0]);
}
